### PR TITLE
Translations: Added a section on general use in PHP

### PR DESCRIPTION
**Pull request by Josniii**
I noticed there was a section on using translation inside PHP controllers, but nothing noted about general use, so I figured I'd add it, so it's more straight forward if your use-case isn't translating directly in a controller. I spent a bit of time looking for how to do this the first time I needed to do translations in PHP, so I hope this can help new developers @ WEXO and elsewhere save some time getting up to speed on translation usage.

If you think it needs changes, feel free to let me know.

### CODE REVIEW

File: guides/plugins/plugins/storefront/add-translations.md [hunk 1]
```diff
@@ -80,3 +80,27 @@ Translation with placeholders:
 $this->trans('soldProducts', ['%count%' => 3, '%country%' => 'Germany']);
 ```
 
+## Using translation generally in PHP
+
+If we need to use a snippet elsewhere in PHP, we can use Dependency Injection to inject `Shopware\Core\Framework\Adapter\Translation\Translator` which implements Symfony's `Symfony\Contracts\Translation\TranslatorInterface`:
+
+```xml
+<service id="Swag\Example\Service\SwagService" public="true" >
+    <argument type="service" id="Shopware\Core\Framework\Adapter\Translation\Translator" />
```
Comment (JoshuaBehrens) [inline guides/plugins/plugins/storefront/add-translations.md:89]: I think you should refer to the `translator` service id instead as the adapter here is just a decorator for that service.

**Review by JoshuaBehrens — CHANGES_REQUESTED** [review]

File: guides/plugins/plugins/storefront/add-translations.md (hunk 1 above)
Comment (Josniii) [reply guides/plugins/plugins/storefront/add-translations.md]: Done. I hadn't even noticed that myself, I've even seen several store plugins that use the Adapter directly.

Comment (JoshuaBehrens) [reply guides/plugins/plugins/storefront/add-translations.md]: 👀 I don't think that this is a good idea but hey good that the docs are better

**Review by JoshuaBehrens — APPROVED** [review]

File: guides/plugins/plugins/storefront/add-translations.md [hunk 2]
```diff
@@ -80,3 +80,27 @@ Translation with placeholders:
 $this->trans('soldProducts', ['%count%' => 3, '%country%' => 'Germany']);
 ```
 
+## Using translation generally in PHP
+
+If we need to use a snippet elsewhere in PHP, we can use Dependency Injection to inject `translator` which implements Symfony's `Symfony\Contracts\Translation\TranslatorInterface`:
```
Comment (PaddyS) [inline guides/plugins/plugins/storefront/add-translations.md:85]:
```suggestion
If we need to use a snippet elsewhere in PHP, we can use [Dependency Injection](../plugin-fundamentals/dependency-injection.md) to inject the `translator`, which implements Symfony's `Symfony\Contracts\Translation\TranslatorInterface`:
```

**Review by PaddyS — CHANGES_REQUESTED** [review]

File: guides/plugins/plugins/storefront/add-translations.md (hunk 1 above)
Comment (Josniii) [reply guides/plugins/plugins/storefront/add-translations.md]: Me neither. But when the core itself uses the Adapter (https://github.com/shopware/core/blob/trunk/Checkout/DependencyInjection/document.xml#L71), that's what you find in the code and end up copying when the docs don't mention it 😄

File: guides/plugins/plugins/storefront/add-translations.md (hunk 2 above)
Comment (Josniii) [reply guides/plugins/plugins/storefront/add-translations.md]: Added, thanks!

**Review by PaddyS — APPROVED** [review]

File: guides/plugins/plugins/storefront/add-translations.md (hunk 1 above)
Comment (JoshuaBehrens) [reply guides/plugins/plugins/storefront/add-translations.md]: Will you or do I PR that away?

Comment (Josniii) [reply guides/plugins/plugins/storefront/add-translations.md]: I found it quite a few places, I'll make a PR that removes it 😄

Comment (JoshuaBehrens) [reply guides/plugins/plugins/storefront/add-translations.md]: Link it to me :D

Comment (Josniii) [reply guides/plugins/plugins/storefront/add-translations.md]: PR is here:
https://github.com/shopware/platform/pull/1884

**Comment by PaddyS** [comment]
Thank you very much @Josniii 🎉

**Pull request merged**